lib/rubocop/cop/rails/duplicate_association.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # This cop looks for associations that have been defined multiple times in the same file.
+      #
+      # @example
+      #
+      #   # bad
+      #   belongs_to :foo
+      #   belongs_to :bar
+      #   has_one :foo
+      #
+      #   # good
+      #   belongs_to :foo
+      #   has_one :bar
+      #
+      class DuplicateAssociation < Base
+        include RangeHelp
+        extend AutoCorrector
+
+        MSG = "Association `%<name>s` is defined multiple times. Don't repeat associations in a model."
+
+        def_node_matcher :association, <<~PATTERN
+          (send nil? ${:belongs_to :has_one :has_many :has_and_belongs_to_many} ({sym str} $_) ...)
+        PATTERN
+
+        def on_class(class_node)
+          offenses(class_node).each do |name, nodes|
+            length = nodes.length
+
+            nodes.each.with_index do |node, i|
```

Comment:
Maybe `nodes[0..-2]` and remove `length`/`with_index`/`i`/`if`?

Reply:
Please disregard this. I've missed the fact that we raise offences for all, but remove all except the last one on autocorrection.

Alternatively,
![image](https://user-images.githubusercontent.com/6916/147556086-46ca6868-070b-4ec1-bee8-8f076151b0b9.png)

Leaving this up to you if you'd like to keep this as is.

Reply:
Nice! That line's too long. Moving it to a guard clause.